### ISSUE TITLE
Test snapshot packages for v0.10.0

### ISSUE THREAD

**Issue by amarthadan** [api3dao/airnode#1633]
Prior to the v0.10.0 release, the snapshot packages and corresponding Docker images need to be tested.

NPM packages:
```
@api3/airnode-abi@snapshot-v0.10.0
@api3/airnode-adapter@snapshot-v0.10.0
@api3/airnode-admin@snapshot-v0.10.0
@api3/airnode-deployer@snapshot-v0.10.0
@api3/airnode-examples@snapshot-v0.10.0
@api3/airnode-node@snapshot-v0.10.0
@api3/airnode-operation@snapshot-v0.10.0
@api3/airnode-protocol@snapshot-v0.10.0
@api3/airnode-utilities@snapshot-v0.10.0
@api3/airnode-validator@snapshot-v0.10.0
```

Docker images:
```
api3/airnode-deployer-dev:snapshot-v0.10.0
api3/airnode-client-dev:snapshot-v0.10.0
api3/airnode-admin-dev:snapshot-v0.10.0
```

**Comment by dcroote** [comment]
## Snapshot testing

**tl;dr** - good to go, except the standalone `airnode-deployer` npm package behavior is questionable.

**A checked box indicates success.**

Test that npm packages `admin`, `deployer`, and `validator`, which have a `--help` CLI argument, are compiled correctly using `npx`

- [x] admin
  - [x] `generate-mnemonic`
- [x] deployer (tested more below)
- [x] validator
  - [x] Succeeds with `--config "packages/airnode-validator/test/fixtures/config.valid.json" --secrets "packages/airnode-validator/test/fixtures/secrets.valid.env"`

Prepare for deployments (needs to be repeated for local, AWS, and GCP):

- [x] Modify the `coingecko-signed-data` integration in `airnode-examples` so that blockchain, HTTP gateway, and HTTP signed gateway requests can be performed. (Aside: not having a single integration that has both the HTTP gateway and signed HTTP gateway enabled is a bit annoying).

Run Airnode client Docker image locally:

- [x] Local blockchain request
- [x] HTTP gateway request
- [x] Signed HTTP gateway request
- [x] Withdrawal request

Test AWS deployment with `deployer` Docker image:

- Requests
  - [x] Testnet request
  - [x] HTTP gateway request
  - [x] Signed HTTP gateway request
  - [x] Withdrawal request
- Deployer commands (with Docker image)
  - [x] `list`
  - [x] `info`
  - [x] `remove`

Test GCP deployment with `deployer` Docker image:

- Requests
  - [x] Testnet request
  - [x] HTTP gateway request
  - [x] Signed HTTP gateway request
  - [x] Withdrawal request
- Deployer commands (with Docker image)
  - [x] `list`
  - [x] `info`
  - [x] `remove-with-receipt` (differs from AWS to achieve additional command coverage) - I had to run this multiple times due to connectivity issues, but I believe that was transient.

## Failures

Any command with the standalone `airnode-deployer` **npm package** e.g. `deploy`, `info`, `remove`, etc. failed. I believe the issue is that the npm package is not loading credentials e.g. `gcp.json` or `aws.env` appropriately like the docker image does; the [deployer package docs](https://docs.api3.org/airnode/v0.10/reference/packages/deployer.html) have no mention of credential files. However, I'm not sure if this is a bug or not given our precautionary wording at the top of the docs and the fact that the package works fine within the deployer docker image, which is what we recommend users use.

@amarthadan is this expected? If so, I find it strange to publish a package that is completely unusable on its own...

Example error when trying to use `list` with a gcp deployment (I had a copy of `gcp.json` in both the directory I was running the command from and in a `config` subdirectory):

```sh
✖ Failed to fetch deployments from GCP: Error: Failed to list GCS buckets: Error: Could not load the default credentials. Browse to https://cloud.google.com/docs/authentication/getting-started for more information.
```

**Comment by amarthadan** [comment]
The `airnode-deployer` NPM package does work but you need to load the credentials in the environment yourself. This is done by the Docker container itself not by the `airnode-deployer`.

For AWS, you have two options. Either set all the AWS-related environment variables (`AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, and possibly `AWS_SESSION_TOKEN`) or populate `~/.aws/credentials` file with valid credentials.

For GCP, you need to set `GOOGLE_APPLICATION_CREDENTIALS` environment variable so it contains a path to your `gcp.json`.

But you're right, that should be mentioned in the docs.

**Comment by bbenligiray** [comment]
Created the docs issue

**Comment by dcroote** [comment]
Closing as complete after confirming that exporting `GOOGLE_APPLICATION_CREDENTIALS` for gcp and the combination of `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` for aws result in working `airnode-deployer` npm package commands.
